From e44e1064ccec2aa09fc66bd750d66919129ae6b4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Przemys=C5=82aw=20Szczepaniak?= <pszczepaniak@google.com>
Date: Fri, 17 Mar 2023 16:01:05 +0000
Subject: [PATCH] Fix OOB read in parseInputs in ShimPreparedModel.cpp

Bug: 269455813
Test: built image
(cherry picked from https://android-review.googlesource.com/q/commit:404c6882edc91a34f60cd56f724d745b06a73756)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:553bc285b36b6312b24087bd085f12620028fac8)
Merged-In: Ib93baacd45f394568eb789b5196a5d0b4e1edd41
Change-Id: Ib93baacd45f394568eb789b5196a5d0b4e1edd41
---
 shim_and_sl/ShimPreparedModel.cpp | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/shim_and_sl/ShimPreparedModel.cpp b/shim_and_sl/ShimPreparedModel.cpp
index 178cc1c3..840d65a1 100644
--- a/shim_and_sl/ShimPreparedModel.cpp
+++ b/shim_and_sl/ShimPreparedModel.cpp
@@ -85,6 +85,11 @@ ErrorStatus ShimPreparedModel::parseInputs(
     }
 
     const auto& model = mMainAndReferencedModels[0];
+
+    if (request.inputs.size() > model.getInputs().size()) {
+        return ErrorStatus::INVALID_ARGUMENT;
+    }
+
     // set inputs
     for (int i = 0; i < request.inputs.size(); ++i) {
         const auto& input = request.inputs[i];
@@ -107,6 +112,9 @@ ErrorStatus ShimPreparedModel::parseInputs(
         }
     }
 
+    if (request.outputs.size() > model.getOutputs().size()) {
+        return ErrorStatus::INVALID_ARGUMENT;
+    }
     // set outputs
     for (int i = 0; i < request.outputs.size(); ++i) {
         const auto& output = request.outputs[i];
-- 
2.41.0.255.g8b1d071c50-goog

